### PR TITLE
Python3 compatibility

### PR DESCRIPTION
**Pull request by FlogFr**
Upgrade CI + library to make it python 3 compatible
- [x] make CI under py2 + py3
- [x] make unittests pass under py2 + py3

### CODE REVIEW

File: circle.yml
Comment (k4nar) [inline circle.yml:3]: You need that because of tox ? I think circle.ci handles it automatically.

Comment (FlogFr) [reply circle.yml:3]: We need that because tox doesn't know which version of python 3.4 / 3.5 it needs to use. The basic setup of the ubuntu 14.04 on Circle CI doesn't link python 3.5 to python 3.5.1 and 3.4 to 3.4.4 etc.
So adding this line is like adding a link for python 3.5.1 to python3.5, so tox knows which python to use.

Comment (k4nar) [reply circle.yml:3]: Ok nice :)

**Review by k4nar — APPROVED** [review]
I think we should wait for a review from @toopy though.

**Review by toopy — APPROVED** [review]

**Pull request merged**
